webapp/channels/src/components/admin_console/ldap_wizard/ldap_wizard.tsx

```diff
@@ -0,0 +1,600 @@
+                state={state}
+                onChange={handleChange}
+                schema={schema}
+                disabled={isDisabled(setting)}
+                setByEnv={isSetByEnv(setting.key!, props.environmentConfig)}
+                setting={setting}
+            />
+        );
+    };
+
+    const buildButtonSetting = (setting: AdminDefinitionSetting) => {
+        return (
+            <LDAPButtonSetting
+                {...props}
+                key={schema.id + '_button_' + setting.key}
+                onChange={handleChange}
+                saveNeeded={false}
+                schema={schema}
+                disabled={isDisabled(setting)}
+                setting={setting as AdminDefinitionSettingButton}
+            />
+        );
+    };
+
+    const buildJobsTableSetting = (setting: AdminDefinitionSetting) => {
+        return (
+            <LDAPJobsTableSetting
+                {...props}
+                key={schema.id + '_jobstable_' + setting.key}
+                schema={schema}
+                disabled={isDisabled(setting)}
+                setting={setting}
+            />
+        );
+    };
+
+    const fileUploadSetstate = (key: string, filename: string | null, errorMessage: string | null) => {
+        setState((prev) => ({
+            ...prev,
+            [key]: filename,
+            [key + 'Error']: errorMessage,
+        }));
+    };
+
+    const buildFileUploadSetting = (setting: AdminDefinitionSetting) => {
+        return (
+            <LDAPFileUploadSetting
+                {...props}
+                key={schema.id + '_fileupload_' + setting.key}
+                value={state[setting.key!]}
+                onChange={handleChange}
+                fileUploadSetstate={fileUploadSetstate}
+                schema={schema}
+                disabled={isDisabled(setting)}
+                setByEnv={isSetByEnv(setting.key!, props.environmentConfig)}
+                setting={setting as AdminDefinitionSettingFileUpload}
+            />
+        );
+    };
+
+    const buildCustomSetting = (setting: AdminDefinitionSetting) => {
+        return (
+            <LDAPCustomSetting
+                {...props}
+                key={schema.id + '_custom_' + setting.key}
+                schema={schema}
+                setting={setting}
+                value={state[setting.key!]}
+                disabled={isDisabled(setting)}
+                setByEnv={isSetByEnv(setting.key!, props.environmentConfig)}
+                onChange={handleChange}
+                registerSaveAction={registerSaveAction}
+                setSaveNeeded={setSaveNeeded}
+                unRegisterSaveAction={unRegisterSaveAction}
+                cancelSubmit={cancelSubmit}
+                showConfirmId={state.showConfirmId}
+            />
+        );
+    };
+
+    // To satisfy type checking
+    const nullFunction = () => {
+        return null;
+    };
+
+    const buildSettingFunctions = {
```

Comment:
Just a thought, but I wonder if this would be easier to follow if we extracted a helper that generated this map and all the functions inside to reduce the size of the component. Given those need the "methods" of this component though, that's probably not actually that helpful even if it would reduce the line count of the `LdapWizard` a bit